src/battle_util.c

```diff
@@ -4695,6 +4703,17 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
                 effect++;
             }
             break;
+        case ABILITY_ZERO_TO_HERO:
+            if (!gSpecialStatuses[battler].switchInAbilityDone
+             && gBattleMons[battler].species == SPECIES_PALAFIN_HERO
```

Comment:
Should be checking the party mon species. Doing it the current way causes the string to be played when this is acquired via Transform.

Reply:
Ah, didn't consider that case. @AlexOn1ine can you make a test and see if using Transform triggers Zero to Hero?

Reply:
> Should be checking the party mon species. Doing it the current way causes the string to be played when this is acquired via Transform.

Transform seemed fine but Imposter did play the message. 
Hopefully fixed in #3552. Thank you!

Reply:
Slightly concerning that the message didn't play for Transform but it did for Imposter. Leads me to wonder if other switch-in Abilities aren't triggering after Transform is used.

Reply:
It's possible, we haven't added Transform tests, so that's something to go back and check.